controllers/servicebinding_controller.go

```diff
@@ -633,13 +633,24 @@ func (r *ServiceBindingReconciler) validateSecretNameIsAvailable(ctx context.Con
 	if err != nil {
 		return client.IgnoreNotFound(err)
 	}
-	if otherBindingName, ok := currentSecret.Labels["binding"]; otherBindingName != binding.Name || !ok {
-		if len(otherBindingName) > 0 {
-			return fmt.Errorf("secret %s belongs to another binding %s, choose a differnet name", binding.Spec.SecretName, otherBindingName)
+
+	if metav1.IsControlledBy(currentSecret, binding) {
+		return nil
+	}
+
+	ownerRef := metav1.GetControllerOf(currentSecret)
+	if ownerRef != nil {
+		owner, err := schema.ParseGroupVersion(ownerRef.APIVersion)
+		if err != nil {
+			return err
+		}
+
+		if owner.Group == binding.GroupVersionKind().Group && ownerRef.Kind == binding.Kind {
+			return fmt.Errorf("secret %s belongs to another binding %s, choose a differnet name", binding.Spec.SecretName, ownerRef.Name)
 		}
-		return fmt.Errorf("the specified secret name '%s' is already taken. Choose another name and try again", binding.Spec.SecretName)
 	}
-	return nil
+
+	return fmt.Errorf("the specified secret name '%s' is already taken. Choose another name and try again", binding.Spec.SecretName)
```

Comment:
Shouldn't that be inside the `ownerRef != nil`? So that this method returns `nil` when there is no owner?